src/value/owned.rs

```diff
@@ -62,6 +74,24 @@ impl OwnedSymbolToken {
     }
 }
 
+impl PartialEq for OwnedSymbolToken {
+    fn eq(&self, other: &Self) -> bool {
+        if other.text != None && self.text != None {
+            if other.text == self.text {
+                return true;
+            }
+            false
+        } else {
+            if self.source == other.source {
+                return true;
+            }
+            false
+        }
+    }
+}
```

Comment:
I might make this a bit more concise without the extra returns and explicit Boolean values

```rust
        if other.text != None && self.text != None {
            // both has text
            other.text == self.text
        } else {
            // one of the tokens has undefined text, the only way they can be equal if they have the same source
            // None implies $0
            self.source == other.source
        }
```

Note that we probably should document the `None` implies `$0` bit in our APIs (if you agree), and we really need tests for the different cases described in:

https://amzn.github.io/ion-docs/docs/symbols.html#data-model

Reply:
Oh yes! I will change it to be more concise. And yes, I think we should document None implies $0. I will try to add tests for above mentioned cases.